src/common/S3Integrations.jsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import { useEffect, useState } from 'react'
+import ExternalLink from './ExternalLink'
+import LoadingIcon from './LoadingIcon'
+
+/**
+ * Provides the last update date string of an S3 file
+ * @param {String} url S3 file
+ * @param {Boolean} shouldFetch Workaround to skip fetching of file headers
+ * @returns Date string OR null
+ */
+export const useS3LastUpdated = (url, shouldFetch) => {
```

Comment:
just a thought for future improvement? Is it worth to store these information in redux? so we wouldn't constantly make these requests whenever we navigate away and back into the page?

Reply:
👍🏽 Added #1260 to follow-up on this.  Thanks!